streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala

```diff
@@ -508,47 +519,7 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
     joiner: (V, VO) => VR,
     windows: JoinWindows
   )(implicit streamJoin: StreamJoined[K, V, VO]): KStream[K, VR] =
-    inner.join[VO, VR](otherStream.inner, joiner.asValueJoiner, windows, streamJoin)
```

Comment:
Just some reordering of methods... First all stream-stream, than all stream-table, than all stream-globalTable joins